site/src/main/java/uk/nhs/digital/ps/beans/navigation/CiBreadcrumbProvider.java

```diff
@@ -0,0 +1,122 @@
+package uk.nhs.digital.ps.beans.navigation;
+
+import org.hippoecm.hst.container.RequestContextProvider;
+import org.hippoecm.hst.content.beans.standard.HippoBean;
+import org.hippoecm.hst.content.beans.standard.HippoFolder;
+import org.hippoecm.hst.core.component.HstRequest;
+import org.hippoecm.hst.core.request.HstRequestContext;
+import org.onehippo.forge.breadcrumb.om.BreadcrumbItem;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.nhs.digital.ps.beans.*;
+
+import java.util.*;
+
+/**
+ * <p>
+ * Default Hippo Breadcrumb plugin uses menu/folder structure to generate breadcrumbs,
+ * but this doesn't fit for clinical indicator content (ciLanding, publication,
+ * series, datasets) as they have a very specific setup in the CMS.
+ * </p>
+ */
+public class CiBreadcrumbProvider  {
+
+    private static final Logger log = LoggerFactory.getLogger(CiBreadcrumbProvider.class);
+    private static final String PATH_CI_LANDING_PAGES = "publication-system/ci-hub/";
+    private static final String SEPARATOR = "/";
+
+    private static final Map<String, String> ciFolderToCiLandingMap = new HashMap<String, String>() {
+        {
+            put("ccg-outcomes-indicator-set", "ccg-outcomes-indicator-set");
+            put("compendium-of-population-health-indicators", "compendium-indicators");
+            put("adult-social-care-outcomes-framework-ascof", "social-care");
+            put("nhs-outcomes-framework", "nhs-outcomes-framework");
+            put("seven-day-services", "seven-day-services");
+            put("shmi", "summary-hospital-level-mortality-indicator-shmi");
+            put("uk-gtn", "uk-gtn");
+        }
+    };
+
+    public CiBreadcrumb getBreadcrumb(final HstRequest request) {
+
+        final HstRequestContext ctx = request.getRequestContext();
+        HippoBean content = ctx.getContentBean();
+        boolean isClinicalIndicatorDocument = false;
+        List<BreadcrumbItem> ciBreadcrumbItems = new ArrayList<>();
+
+        // Is this a CiLanding page?
+        if (ciFolderToCiLandingMap.containsValue(content.getName())) {
+            isClinicalIndicatorDocument = true;
+        } else {
```

Comment:
A lot of logic happens in this `else` block, I wonder if it could be made a bit more easy to follow if a few blocks were moved to private methods:
* the entire `while` loop (together with the preceding `HippoFolder folder` line, moved to something like `addCiLandingPageBreadcrumbItem(List<BreadcrumbItem> ciBreadcrumbItems)`,
* content of `if (content.getClass().equals(Publication.class)) {` block moved to, say, `addPublicationBreadcrumbItem(List<BreadcrumbItem> ciBreadcrumbItems)`,
* content of `} else if (content.getClass().equals(Dataset.class)) {` block moved to `addDatasetBreadcrumbItem(List<BreadcrumbItem> ciBreadcrumbItems)`

...thus applying the [SLAP principle](https://hackernoon.com/object-oriented-tricks-6-slap-your-functions-a13d25a7d994).

Reply:
Good call.  Reads a lot clearer now 👍